### PR TITLE
fix: Support Indy-style identifiers [DEV-1750]

### PR DESCRIPTION
**Pull request by Toktar**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-1750 Change identifier validation to support Indy-style identifiers correctly](https://app.clickup.com/t/6600954/DEV-1750)

File: x/cheqd/utils/did.go [hunk 1]
```diff
@@ -84,8 +84,7 @@ func IsValidDID(did string, method string, allowedNamespaces []string) bool {
 }
 
 func ValidateID(id string) error {
-	isValidId := len(id) == 16 && IsValidBase58(id) ||
-		len(id) == 32 && IsValidBase58(id) ||
+	isValidId := len(id) >= 16 && len(id) <= 32 && IsValidBase58(id) ||
```
[the remark this answers is not in the document]
Comment (Toktar) [reply x/cheqd/utils/did.go:87]: Maybe I missed something... But you have approved this approach
https://github.com/cheqd/node-docs/pull/15

File: x/cheqd/utils/did_url_test.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ func TestIsDidURL(t *testing.T) {
 		{"Valid: the whole alphabet with query and apth", true, "did:cheqd:testnet:123456789abcdefg/path?query#abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~!$&'()*+,;=:@%20%ff?"},
 		{"Valid: only fragment", true, "did:cheqd:testnet:123456789abcdefg#fragment"},
 		{"Valid: only query", true, "did:cheqd:testnet:123456789abcdefg?query"},
+		{"Valid: UniqueID more then 16 symbols and less then 32", true, "did:cheqd:testnet:123456789abcdefgABCDEF/path?query#fragment"},
```
[the remark this answers is not in the document]
Comment (Toktar) [reply x/cheqd/utils/did_url_test.go:30]: Just utils, but got your point.  I'll fix it

File: x/cheqd/utils/did.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (askolesov) [reply x/cheqd/utils/did.go:87]: Should we do this validation **exactly** as in Indy? `len(decodeBase58(identifier string)) == 16..32`

[the remark this answers is not in the document]
Comment (Toktar) [reply x/cheqd/utils/did.go:87]: We already have the main cheqd DID Doc with 32 symbol identifier. Do we want to deprecate it?

[the remark this answers is not in the document]
Comment (Toktar) [reply x/cheqd/utils/did.go:87]: `did:cheqd:mainnet:zF7rhDBfUt9d1gJPjx7s1JXfUY7oVWkY`

[the remark this answers is not in the document]
Comment (askolesov) [reply x/cheqd/utils/did.go:87]: @ankurdotb It took me some time to understand the issue.
```
16-32 bytes in base58 ~= 22-44 chars
```
So roughly saying we gonna change allowed range in chars:
```
16-32 -> 22-44
```
And @Toktar 's question is what to do with dids of length `16-22` that were valid previously, will become invalid after the change?

**Comment by Toktar** [comment]
@ankurdotb I updated test and answered with an option list in our conversation. Could you please take a look when having time?

File: x/cheqd/utils/did.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankurdotb) [reply x/cheqd/utils/did.go:87]: Can't we do Option 1 and truncate existing Indy-style identifiers to what's expected in the Indy spec? We're changing protobufs and therefore will need to migrate existing DIDs/DIDDOcs anyway

[the remark this answers is not in the document]
Comment (Toktar) [reply x/cheqd/utils/did.go:87]: Do we want to change/remove an identifier of Cheqd main DID which was published in many social messages on the Internet? `did:cheqd:mainnet:zF7rhDBfUt9d1gJPjx7s1JXfUY7oVWkY`

**Comment by ankurdotb** [comment]
@askolesov @Eengineer1 @anikitinDSR This needs to have changes from develop merged in, tests have changed

**Review by askolesov — DISMISSED** [review]
Perfect, thanks

File: x/cheqd/utils/did.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (askolesov) [reply x/cheqd/utils/did.go]: So we decided to go with option 1

**Review by askolesov — APPROVED** [review]
Nice

**Pull request merged**